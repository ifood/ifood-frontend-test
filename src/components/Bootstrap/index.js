import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Media from 'react-bootstrap/Media'
import Modal from 'react-bootstrap/Modal'
import ModalDialog from 'react-bootstrap/ModalDialog'
import ModalHeader from 'react-bootstrap/ModalHeader'
import ModalTitle from 'react-bootstrap/ModalTitle'
import ModalBody from 'react-bootstrap/ModalBody'

export {
  Container,
  Row,
  Col,
  Media,
  Modal,
  ModalDialog,
  ModalHeader,
  ModalTitle,
  ModalBody,
}
