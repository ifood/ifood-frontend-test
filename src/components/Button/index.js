import React from "react";
import PropTypes from "prop-types";

import * as S from "./styled";

const Button = ({ onClick, label, variation }) => {
  return (
    <S.Button variation={variation} onClick={onClick}>
      {label}
    </S.Button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  variation: PropTypes.oneOf(["primary", "secondary"]).isRequired,
};

export default Button;
