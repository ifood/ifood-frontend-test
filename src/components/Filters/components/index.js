/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import Amount from './Amount';
import DatePicker from './DatePicker';
import Page from './Page';
import SelectCountry from './SelectCountry';
import SelectLocale from './SelectLocale';

const locale = (props) => <SelectLocale {...props} />;
const timestamp = (props) => <DatePicker {...props} />;
const offset = (props = {}) => <Page {...props} />;
const country = (props) => <SelectCountry {...props} />;
const limit = (props) => <Amount {...props} />;

export default { locale, timestamp, offset, country, limit };
