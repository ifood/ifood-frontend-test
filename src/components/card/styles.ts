import styled from 'styled-components';
import breakpoints from 'common/breakpoints';

export const Card = styled.div`
  background: #ffffff;
  border: 1px solid #e5ecff;
  box-sizing: border-box;
  border-radius: 4px;
  margin-top: 20px;
  flex: 1 0 100%;
  margin: 5px;
  transition: all 0.1s ease-in-out;
  min-width: 200px;
  max-width: 300px;
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14),
    0px 1px 3px 0px rgba(0, 0, 0, 0.12);

  &:hover {
    transform: scale(1.1);
    box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 1px 4px 0px rgba(0, 0, 0, 0.14),
      0px 1px 9px 0px rgba(0, 0, 0, 0.12);
  }

  @media (${breakpoints.MEDIUM}) {
    flex: 1 0 40%;
  }
`;
