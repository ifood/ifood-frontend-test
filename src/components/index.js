import Loading from './loading'
import Header from './header'
import Filters from './filters'
import CardFeatured from './card-featured'
import Premium from './premium'

export {
  Loading,
  Header,
  Filters,
  CardFeatured,
  Premium
}
