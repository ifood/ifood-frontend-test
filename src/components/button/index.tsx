import React, { ButtonHTMLAttributes } from 'react'; //eslint-disable-line

import * as S from './styles';

interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
}

export const Button: React.FC<IButton> = ({ children, ...rest }) => {
  return <S.Button {...rest}>{children}</S.Button>;
};
