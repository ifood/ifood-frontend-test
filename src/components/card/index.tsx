import React from 'react'; //eslint-disable-line

import * as S from './styles';

interface ICard {
  children: React.ReactNode;
}

export const Card: React.FC<ICard> = ({ children, ...rest }) => {
  return <S.Card {...rest}>{children}</S.Card>;
};
