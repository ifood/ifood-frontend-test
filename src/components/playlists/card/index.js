import React from 'react';
import PropTypes from 'prop-types';

import * as S from './styles';

function Card({
    background,
    description,
    href,
 }) {
    const _openPlaylist = () => (
        window.open(href, '_blank')
    );

    return (
        <S.Container onClick={_openPlaylist}>
            <S.Banner src={background} />
            <S.Description>{ description }</S.Description>
        </S.Container>
    );
}

Card.propTypes = {
    background: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
};

export default Card;