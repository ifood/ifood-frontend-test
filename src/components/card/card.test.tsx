import React from 'react';
import { render, screen } from '@testing-library/react';
import { Card } from 'components/card';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

const wrapper = () => render(<Card>Card text</Card>);

test('testing accessibility', async () => {
  const { container } = wrapper();
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});

test('renders component without errors', () => {
  const { container } = wrapper();

  expect(screen.getByText(/Card text/)).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
