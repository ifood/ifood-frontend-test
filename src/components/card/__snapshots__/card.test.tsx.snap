// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component without errors 1`] = `
<div>
  <div
    class="sc-bdnylx kIxwpW"
  >
    Card text
  </div>
</div>
`;
